### PR TITLE
WTBUILD-106 Create a delete function to manage snapshots in fabfile

### PR DESCRIPTION
**Pull request by ruby-oujo**

### CODE REVIEW

File: fabfile.py [hunk 1]
```diff
@@ -407,6 +407,29 @@ def workload(c, upload=None, describe=None):
     
     return current_workload or None
 
+# The function will take a specified snapshot_id and delete the corresponding snapshot.
```
Comment (etienneptl) [inline fabfile.py:410]: We don't need to name variables in a comment:
```suggestion
# The function will take a specified snapshot ID and delete the corresponding snapshot.
```

File: fabfile.py [hunk 2]
```diff
@@ -407,6 +407,29 @@ def workload(c, upload=None, describe=None):
...
+@task
+def snapshot_delete(c, snapshot_id=None):
```
Comment (etienneptl) [inline fabfile.py:412]: Please make the arg mandatory.
```suggestion
def snapshot_delete(c, snapshot_id):
```
You can then remove the code just below.

File: fabfile.py (hunk 1 above)
Comment (etienneptl) [inline fabfile.py:410]: Please explain that a list can be provided.

File: fabfile.py [hunk 3]
```diff
@@ -407,6 +407,29 @@ def workload(c, upload=None, describe=None):
...
+        print("Please specify the snapshot(s) you wish to delete through \
+              --snapshot_id=<snapshotid,snapshotid1> separated by a ',' .")
+    else:
+        if "," in snapshot_id:
+            snapshot_ids = snapshot_id.split(",")
+            for snapshot_id in snapshot_ids:
+                result = c.run("aws ec2 delete-snapshot \
+                --snapshot-id {snapshot_id}")
+                if result.stderr:
+                    raise Exit(result.stderr)
```
Comment (etienneptl) [inline fabfile.py:423]: Should we print the error but still try to delete the other snapshots?

File: fabfile.py [hunk 4]
```diff
@@ -407,6 +407,29 @@ def workload(c, upload=None, describe=None):
...
+                print(f"Deleted snapshot '{snapshot_id}'.")        
+
+        else:
```
Comment (etienneptl) [inline fabfile.py:426]: I think you should be able to create a list of snapshots if only one is provided and use the code above.

**Review by etienneptl — CHANGES_REQUESTED** [review]

File: fabfile.py (hunk 3 above)
Comment (ruby-oujo) [reply fabfile.py:423]: I think it is risky to continue deleting when one is wrong.

Comment (etienneptl) [reply fabfile.py:423]: Yes, that's fair.

File: fabfile.py (hunk 2 above)
Comment (ruby-oujo) [reply fabfile.py:412]: This allows me to give a more descriptive error message

File: fabfile.py (hunk 1 above)
Comment (ruby-oujo) [reply fabfile.py:410]: done

File: fabfile.py (hunk 4 above)
Comment (ruby-oujo) [reply fabfile.py:426]: done

File: fabfile.py (hunk 2 above)
Comment (etienneptl) [reply fabfile.py:412]: As discussed, let's leave it.

**Review by etienneptl — APPROVED** [review]
LGTM!

**Pull request merged**